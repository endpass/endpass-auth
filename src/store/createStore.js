// @ts-check
import Vue from 'vue';
import Vuex from 'vuex';

Vue.use(Vuex);

/**
 * @param {object?} [options] passed options for store
 * @return {import('vuex').Store<{}>} store Vuex store instance
 */
export default options => {
  return new Vuex.Store({
    strict: !ENV.VUE_APP_IS_PRODUCTION,
    ...options,
  });
}
