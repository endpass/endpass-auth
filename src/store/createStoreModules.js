import GasPriceModule from '@/store/modules/GasPriceModule';
import SharedModule from '@/store/modules/SharedModule';
import ChannelStore from '@/store/modules/ChannelModule';
import BalanceModule from '@/store/modules/BalanceModule';
import AccountsModule from '@/store/modules/AccountsModule';
import AuthModule from '@/store/modules/AuthModule';
import CoreModule from '@/store/modules/CoreModule';
import RequestsModule from '@/store/modules/RequestsModule';
import moduleFactory from '@/store/moduleFactory';

export default store => {
  const createModule = moduleFactory(store);

  const gasPriceStore = createModule(GasPriceModule, 'gasPrice');

  const sharedStore = createModule(SharedModule, 'shared');

  const channelStore = createModule(ChannelStore, 'channel');

  const balanceStore = createModule(BalanceModule, 'balance');

  const accountsStore = createModule(AccountsModule, 'accounts', {
    sharedStore,
    balanceStore,
  });

  const authStore = createModule(AuthModule, 'auth', {
    sharedStore,
  });

  const coreStore = createModule(CoreModule, 'core', {
    authStore,
    sharedStore,
  });

  const requestStore = createModule(RequestsModule, 'requests', {
    accountsStore,
    sharedStore,
  });

  return {
    channelStore,
    gasPriceStore,
    sharedStore,
    accountsStore,
    balanceStore,
    authStore,
    coreStore,
    requestStore,
  };
};
