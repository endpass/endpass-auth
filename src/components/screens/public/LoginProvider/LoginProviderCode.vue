<template>
  <message
    v-if="!hasLoginChallenge"
    :error="true"
    data-test="error-message"
  >
    {{ $t('components.loginProviderPassword.loginChallenge') }}
  </message>
  <code-form
    v-else
    :email="email"
    password=""
    :is-sign-up="false"
    :is-closable="false"
    :is-otp="isOtp"
    :submit-handler="authWithCode"
  />
</template>

<script>
import Message from '@/components/common/Message';
import { accountsStore } from '@/store';
import CodeForm from '@/components/forms/Code';
import createLoginController from './LoginController';

export default {
  name: 'LoginProvider',
  accountsStore,

  loginController: createLoginController(),

  props: {
    loginChallenge: {
      type: String,
      default: '',
    },

    email: {
      type: String,
      default: null,
    },
  },

  data: () => ({
    isLoading: false,
  }),

  computed: {
    hasLoginChallenge() {
      return !!this.loginChallenge;
    },

    isOtp() {
      return this.$options.accountsStore.isAppOtp;
    },

    challengeType() {
      return this.$options.accountsStore.challengeType;
    },
  },

  methods: {
    async authWithCode({ code }) {
      await this.$options.loginController.authWithCode({
        challengeId: this.loginChallenge,
        code,
      });
    },
  },

  components: {
    CodeForm,
    Message,
  },
};
</script>
