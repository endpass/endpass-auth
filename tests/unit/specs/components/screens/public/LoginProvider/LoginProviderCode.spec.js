import Vuex from 'vuex';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import '@mocks/window';
import LoginProviderCode from '@/components/screens/public/LoginProvider/LoginProviderCode';
import setupI18n from '@/locales/i18nSetup';
import permissionsService from '@/service/permissions';
import createStore from '@/store/createStore';
import createStoreModules from '@/store/createStoreModules';

const localVue = createLocalVue();

localVue.use(Vuex);
const i18n = setupI18n(localVue);

describe('LoginProviderCode', () => {
  let wrapper;

  const createWrapper = options => {
    const store = createStore();
    const { accountsStore } = createStoreModules(store);

    return shallowMount(LoginProviderCode, {
      accountsStore,
      localVue,
      i18n,
      ...options,
    });
  };

  beforeEach(() => {
    jest.clearAllMocks();
    window.location.href = jest.fn();

    wrapper = createWrapper();
  });

  describe('render', () => {
    it('should correctly render LoginProviderCode screen', () => {
      wrapper = createWrapper();

      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('behavior', () => {
    it('should show error if challengeId is not in props', () => {
      wrapper = createWrapper();

      expect(wrapper.find('[data-test=error-message]').exists()).toBe(true);
      expect(wrapper.find('sign-password-stub').exists()).toBe(false);
    });

    describe('password submit', () => {
      const code = 'foo';
      const challengeId = 'bar';

      it('should handle password submit and makes hydra login', async () => {
        expect.assertions(1);

        wrapper = createWrapper({
          propsData: {
            loginChallenge: challengeId,
          },
        });
        permissionsService.getLoginDetails.mockResolvedValueOnce({});
        permissionsService.login.mockResolvedValueOnce({
          redirect: 'new/path',
        });

        wrapper.vm.authWithCode(code);

        await global.flushPromises();

        expect(window.location.href).toBe('new/path');
      });
    });
  });
});
