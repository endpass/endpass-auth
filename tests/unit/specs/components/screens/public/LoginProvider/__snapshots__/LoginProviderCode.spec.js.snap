// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginProviderCode render should correctly render LoginProviderCode screen 1`] = `
<message-stub error="true" data-test="error-message">
  You should provide login_challenge param in url, add it and try again!
</message-stub>
`;
